pkg/gvk/gvk.go

```diff
@@ -117,12 +117,21 @@ var typeOrders = func() map[string]int {
 
 // IsLessThan returns true if self is less than the argument.
 func (x Gvk) IsLessThan(o Gvk) bool {
-	indexI := typeOrders[x.Kind]
-	indexJ := typeOrders[o.Kind]
-	if indexI != indexJ {
-		return indexI < indexJ
+	if x.Kind != o.Kind {
+		indexI := typeOrders[x.Kind]
+		indexJ := typeOrders[o.Kind]
+		if indexI != indexJ {
+			return indexI < indexJ
+		}
+		return x.Kind < o.Kind
+	}
+	if x.Group != o.Group {
+		return x.Group != "" && (o.Group == "" || x.Group < o.Group)
+	}
+	if x.Version != o.Version {
+		return x.Version != "" && (o.Version == "" || x.Version < o.Version)
 	}
-	return x.String() < o.String()
+	return false
```

Comment:
It just occurred to me that a less intrusive approach would have been to just swap the usage of `i` and `j` in `fsSlice.Less`). I also didn't add a sort on `fsSlice.mergeOne` after the `append` so individual configs could still be out of order.